alicloud/resource_alicloud_kvstore_instance_test.go

```diff
@@ -0,0 +1,239 @@
+		CheckDestroy: testAccCheckKVStoreInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccKVStoreInstance_vpc,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckKVStoreInstanceExists(
+						"alicloud_kvstore_instance.foo", &instance),
+					resource.TestCheckResourceAttr(
+						"alicloud_kvstore_instance.foo",
+						"instance_class",
+						"redis.master.small.default"),
+					resource.TestCheckResourceAttr(
+						"alicloud_kvstore_instance.foo",
+						"engine_version",
+						"2.8"),
+					resource.TestCheckResourceAttr(
+						"alicloud_kvstore_instance.foo",
+						"instance_type",
+						"Redis"),
+					resource.TestCheckResourceAttr(
+						"alicloud_kvstore_instance.foo",
+						"instance_name",
+						"testAccKVStoreInstance_vpc"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccAlicloudKVStoreInstance_upgradeClass(t *testing.T) {
+	var instance r_kvstore.DBInstanceAttribute
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+
+		// module name
+		IDRefreshName: "alicloud_kvstore_instance.foo",
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckKVStoreInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccKVStoreInstance_class,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckKVStoreInstanceExists(
+						"alicloud_kvstore_instance.foo", &instance),
+					resource.TestCheckResourceAttr("alicloud_kvstore_instance.foo", "instance_class", "redis.master.small.default"),
+				),
+			},
+
+			resource.TestStep{
+				Config: testAccKVStoreInstance_classUpgrade,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckKVStoreInstanceExists(
+						"alicloud_kvstore_instance.foo", &instance),
+					resource.TestCheckResourceAttr("alicloud_kvstore_instance.foo", "instance_class", "redis.master.mid.default"),
+				),
+			},
+		},
+	})
+
+}
+
+func testAccCheckKVStoreInstanceExists(n string, d *r_kvstore.DBInstanceAttribute) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No DB Instance ID is set")
+		}
+
+		client := testAccProvider.Meta().(*AliyunClient)
+		attr, err := client.DescribeRKVInstanceById(rs.Primary.ID)
+		log.Printf("[DEBUG] check instance %s attribute %#v", rs.Primary.ID, attr)
+
+		if err != nil {
+			return err
+		}
+
+		if attr == nil {
+			return fmt.Errorf("DB Instance not found")
+		}
+
+		*d = *attr
+		return nil
+	}
+}
+
+func testAccCheckKVStoreInstanceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*AliyunClient)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "alicloud_kvstore_instance" {
+			continue
+		}
+
+		ins, err := client.DescribeRKVInstanceById(rs.Primary.ID)
+
+		if ins != nil {
+			return fmt.Errorf("Error DB Instance still exist")
+		}
+
```

Comment:
I think line 170-172 are useless and method DescribeRKVInstanceById can ensure it.

Reply:
It seems like without any update.